File: app/src/main/java/com/example/puyo_base_simulator/ui/home/LoadFieldAdapter.java
package com.example.puyo_base_simulator.ui.home;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.example.puyo_base_simulator.R;

import java.util.List;

public class LoadFieldAdapter extends RecyclerView.Adapter<LoadFieldAdapter.FieldViewHolder> {

    private List<FieldPreview> mFields;
    private FieldSelectedListener mFieldSelectedListener;

    public LoadFieldAdapter(List<FieldPreview> fields){
        super();
        this.mFields = fields;
    }

    public void setFieldSelectedListener(FieldSelectedListener fieldSelectedListener) {
        this.mFieldSelectedListener = fieldSelectedListener;
    }

    @Override
    public FieldViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new FieldViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_field, parent, false));
    }

    @Override
    public void onBindViewHolder(FieldViewHolder holder, final int position) {
        final FieldPreview fieldPreview = mFields.get(position);
        holder.icon.setImageResource(fieldPreview.iconRes);
        holder.fieldID.setText(fieldPreview.content);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if(mFieldSelectedListener != null){
                    mFieldSelectedListener.onFieldSelected(position, fieldPreview);
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return mFields.size();
    }

    static class FieldViewHolder extends RecyclerView.ViewHolder{
        TextView fieldID;
        ImageView icon;

        public FieldViewHolder(View itemView) {
            super(itemView);
            fieldID = itemView.findViewById(R.id.fieldID);
            icon = itemView.findViewById(R.id.icon);
        }
    }

    interface FieldSelectedListener {
        void onFieldSelected(int position, FieldPreview fieldPreview);
    }
}

File: app/src/main/java/com/example/puyo_base_simulator/ui/home/LoadFieldAdapter.kt
package com.example.puyo_base_simulator.ui.home

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.puyo_base_simulator.R
import com.example.puyo_base_simulator.ui.home.LoadFieldAdapter.FieldViewHolder
import java.util.*
import kotlin.text.*

class LoadFieldAdapter(private val mFields: List<FieldPreview>) : RecyclerView.Adapter<FieldViewHolder>() {
    private lateinit var mFieldSelectedListener: (Int, FieldPreview) -> Unit
    fun setFieldSelectedListener(fieldSelectedListener:(Int, FieldPreview) -> Unit) {
        mFieldSelectedListener = fieldSelectedListener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FieldViewHolder {
        return FieldViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_field, parent, false))
    }

    override fun onBindViewHolder(holder: FieldViewHolder, position: Int) {
        val fieldPreview = mFields[position]
        //holder.fieldID.text = String.format(Locale.JAPAN, "seed: %d",fieldPreview.seed)
        holder.allClear.text = if (fieldPreview.allClear) { "全消し" } else { "" }
        holder.numOfPlacement.text = String.format(Locale.JAPAN, "%d手目", fieldPreview.numOfPlacement)
        holder.point.text = String.format(Locale.JAPAN, "%d点", fieldPreview.point)
        holder.fieldCanvas.setField(fieldPreview.content)
        holder.itemView.setOnClickListener {
            mFieldSelectedListener.invoke(position, fieldPreview)
        }
    }

    override fun getItemCount(): Int {
        return mFields.size
    }

    class FieldViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var allClear: TextView = itemView.findViewById(R.id.allClear)
        var numOfPlacement: TextView = itemView.findViewById(R.id.numOfPlacement)
        var point : TextView = itemView.findViewById(R.id.point)
        var fieldCanvas: FieldPreviewCanvas = itemView.findViewById(R.id.fieldCanvas)
    }
}